core/src/main/java/org/infinispan/context/impl/BaseSequentialInvocationContext.java

```diff
@@ -0,0 +1,316 @@
+package org.infinispan.context.impl;
+
+import org.infinispan.commands.VisitableCommand;
+import org.infinispan.commons.CacheException;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.context.SequentialInvocationContext;
+import org.infinispan.interceptors.SequentialInterceptorChain;
+import org.infinispan.interceptors.SequentialInterceptor;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+import org.jboss.logging.MDC;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * This base class implements the {@link org.infinispan.context.SequentialInvocationContext} methods.
+ *
+ * @author Dan Berindei
+ * @since 8.1
+ */
+public abstract class BaseSequentialInvocationContext
+      implements InvocationContext, SequentialInvocationContext {
+   private static final Log log = LogFactory.getLog(BaseSequentialInvocationContext.class);
+   private static final boolean trace = log.isTraceEnabled();
+   // Enable when debugging BaseSequentialInvocationContext itself
+   private static final boolean EXTRA_LOGS = false;
```

Comment:
Don't you want to get the ```EXTRA_LOGS``` from a system property instead? it would avoid to recompile the code and it would remove all the IDEA warnings about "this condition can be simplify to false" :)